lib/generators/setup_generator.rb

```diff
@@ -0,0 +1,94 @@
+require "colorize"
+
+class SetupGenerator < Rails::Generators::Base
+
+  desc <<-EOS
+    This generator prepares applications for API integration:
+      1. Generates config/config.yml file
+      2. Generates .gitignore file
+      3. Includes api_bridge gem in app's Gemfile
+      4. Generates facets file for customization
+      5. Sets up default stylesheet variables file
+  EOS
+
+  def setup_files
+    @this_dir = File.expand_path File.dirname(__FILE__)
+    @app_dir = Rails.root
+
+    msgs = []
+
+    msgs << copy_configs
+    msgs << gitignore
+    msgs << gems
+    msgs << facets
+    msgs << stylesheet
+
+    puts msgs.join("\n")
+
+    Bundler.with_clean_env do
+      run "bundle install"
+    end
+  end
+
+  private
+
+  def copy_configs
+    puts "Copying in configuration files"
+
+    # initializer
+    # NOTE: This could be done with the "initializer" method instead
+    # http://guides.rubyonrails.org/generators.html#initializer
+    FileUtils.cp("#{@this_dir}/templates/config.rb", "#{@app_dir}/config/initializers/config.rb")
+
+    # example config file
+    FileUtils.cp("#{@this_dir}/templates/config.yml", "#{@app_dir}/config/config.example.yml")
+
+    # config file
+    # does not overwrite existing config file
+    if File.exist?("#{@app_dir}/config/config.yml")
+      return "Verify that your config/config.yml file matches the options in config/config.example.yml".cyan
```

Comment:
I think that they won't be identical, because they should be customizing the config file to have their server, so it won't match the config file being copied in from here.  It will only copy if there isn't a file there.  I'll go through and make sure that exist vs non-exist is standardized